src/java.net.http/share/classes/jdk/internal/net/http/Stream.java

```diff
@@ -1547,6 +1590,8 @@ void cancelImpl(final Throwable e, final int resetFrameErrCode) {
             }
         } catch (Throwable ex) {
             Log.logError(ex);
+        } finally {
+            drainInputQueue();
```

Comment:
Hello Daniel, now that the `cancelImpl(...)` itself drains the queue in a `finally` block, do you think we should remove the call to `drainInputQueue` from the finally block of `schedule()` after we call `cancelImpl(...)` there?

Reply:
Good point.